File: WorldFeed/Server/src/WorldFeed.Common/Public/Models/Interfaces/DTO/QueryDTO/IBaseCursorQueryDTO.cs
namespace WorldFeed.Common.Public.Models.Interfaces.DTO.QueryDTO
{
    using System.Collections.Generic;

    public interface IBaseCursorQueryDTO
    {
        long PreviousCursor { get; set; }
        long NextCursor { get; set; }

        string PreviousCursorStr { get; set; }
        string NextCursorStr { get; set; }

        string RawResult { get; set; }

        int GetNumberOfObjectRetrieved();
    }

    public interface IBaseCursorQueryDTO<T> : IBaseCursorQueryDTO
    {
        IEnumerable<T> Results { get; set; }
    }
}

File: Sharebook/Client/src/app/core/Public/Models/Interfaces/DTO/QueryDTO/IBaseCursorQueryDTO.ts
import {InjectionToken} from "@angular/core";
import {IMessageEventDTO} from "../Events/IMessageEventDTO";

export interface IBaseCursorQueryDTO<T = any> {
  previousCursor: number;
  nextCursor: number;

  previousCursorStr: string;
  nextCursorStr: string;

  rawResult: string;

  results?: Array<T>;

  getNumberOfObjectRetrieved(): number;
}

// export interface IBaseCursorQueryDTO<T> extends IBaseCursorQueryDTO {
//   Results: IEnumerable<T>;
// }
